instrumentation/metric/reader.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 the orbs-network-go authors
+// This file is part of the orbs-network-go library in the Orbs project.
+//
+// This source code is licensed under the MIT license found in the LICENSE file in the root directory of this source tree.
+// The above notice should be included in all copies or substantial portions of the software.
+
+package metric
+
+import (
+	"encoding/json"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+func NewReader(endpoint string) (exportedMap, error) {
+	res, err := http.Get(endpoint)
+	if err != nil {
+		return nil, errors.Errorf("MetricReader cannot read data from endpoint '%s'", endpoint)
+	}
+	if res == nil {
+		return nil, err
+	}
+
+	readBytes, err := ioutil.ReadAll(res.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	status := struct {
+		Payload   exportedMap
+	}{}
+	err = json.Unmarshal(readBytes, &status)
+	if err != nil {
+		return nil, err
+	}
+	return status.Payload, nil
+}
+
+func (mr exportedMap) GetAsInt(name string) (int64, bool) {
+	value, found := mr.Get(name)
+	if found {
+		if valueAsInt, ok := value.(int64); ok {
+			return valueAsInt, true
+		} else if valueAsFloat, ok := value.(float64); ok {
+			return int64(valueAsFloat), true
+		}
+	}
+	return 0, false
+}
+
+func (mr exportedMap) Get(name string) (interface{}, bool) {
+	nameParts := deconstructNesting(name)
+	return travelNesting(mr, nameParts)
+}
+
+func deconstructNesting(name string) []string {
+	nameParts := strings.Split(name, ".")
+	nameLen := len(nameParts)
+
+	knownType := isNameAKnownType(nameParts[nameLen-1])
+	if nameLen > 0 && knownType {
```

Comment:
`nameLen` is always positive otherwise you would panic on the previous line no? maybe you want to check that it's greater than 1 here to ensure the resulting name will not be empty?